File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
var kthSmallest = function(root, k) {
    let inans = [];
    inorder(root, inans)
    return inans[k-1];

    function inorder(root, inans) {
        if(root==null) return;
        inorder(root.left, inans);
        inans.push(root.val);
        inorder(root.right, inans);
        return;
    }
};

// 1,2-3,4-5,6

File: 0230-kth-smallest-element-in-a-bst/0230-kth-smallest-element-in-a-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private static int ans = 0;
    private static int c = 0;

    public int kthSmallest(TreeNode root, int k) {
        c = k;
        inorder(root);
        return ans;
    }

    public void inorder(TreeNode root) {
        if(root.left!=null) inorder(root.left);
        c--;
        if(c==0){
            ans = root.val;
            return;
        }
        if(root.right!=null) inorder(root.right);
    }
}